snuba/web/delete_query.py

```diff
@@ -0,0 +1,49 @@
+from typing import Any, Dict
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.clickhouse.formatter.query import format_query
+from snuba.clickhouse.query import Query
+from snuba.datasets.storage import WritableTableStorage
+from snuba.query.conditions import combine_and_conditions
+from snuba.query.data_source.simple import Table
+from snuba.query.dsl import column, equals, in_cond, literal, literals_tuple
+from snuba.query.expressions import Expression
+from snuba.reader import Result
+
+
+def construct_condition(body: Dict[str, Any]) -> Expression:
+    and_conditions = []
+    for col, values in body["columns"].items():
+        if len(values) == 1:
+            exp = equals(column(col), literal(values[0]))
+        else:
+            literal_values = [literal(v) for v in values]
+            exp = in_cond(
+                column(col), literals_tuple(alias=None, literals=literal_values)
+            )
+
+        and_conditions.append(exp)
+
+    return combine_and_conditions(and_conditions)
+
+
+def delete_query(
+    storage: WritableTableStorage, table: str, body: Dict[str, Any]
+) -> Result:
+    cluster_name = storage.get_cluster().get_clickhouse_cluster_name()
+    on_cluster = literal(cluster_name) if cluster_name else None
```

Comment:
In what situation would `storage.get_cluster()` not throw an `UndefinedClickhouseCluster` error, but `cluster_name` is still None?

Reply:
@xurui-c if the cluster name isn't defined then we have a single node cluster, which means there is no replication going on (this is [what we have locally](https://github.com/getsentry/snuba/blob/master/snuba/settings/__init__.py#L83)). In that case we don't want to add the `ON CLUSTER cluster_name` part of the delete query